src/overlay005/field_control.c

```diff
@@ -0,0 +1,1092 @@
+            ov9_0224A800(fieldSystem, input->transitionDir);
+        } else if (Field_CheckMapTransition(fieldSystem, input) == TRUE) {
+            Field_TrySetMapConnection(fieldSystem);
+            return TRUE;
+        }
+    }
+
+    if (input->registeredItem && sub_02069238(fieldSystem) == TRUE) {
+        return TRUE;
+    }
+
+    if (input->menu && sub_0203A9C8(fieldSystem) == TRUE) {
+        Sound_PlayEffect(SEQ_SE_DP_WIN_OPEN);
+        FieldMenu_Init(fieldSystem);
+        return TRUE;
+    }
+
+    return FALSE;
+}
+
+static BOOL Field_CheckSign (FieldSystem *fieldSystem)
+{
+    MapObject *object;
+
+    if (sub_0203CBE0(fieldSystem, &object) == TRUE) {
+        sub_0203E880(fieldSystem, sub_02062960(object), object);
+        return TRUE;
+    }
+
+    int event = sub_0203CB80(fieldSystem, (void *)sub_0203A440(fieldSystem), sub_0203A448(fieldSystem));
+    
+    if (event != 0xffff) {
+        sub_0203E880(fieldSystem, event, NULL);
+        return TRUE;
+    }
+
+    return FALSE;
+}
+
+BOOL FieldInput_Process_Underground (FieldInput *input, FieldSystem *fieldSystem)
+{
+    if (input->unused5 == FALSE && sub_0203F5C0(fieldSystem, 1) == TRUE) {
+        return TRUE;
+    }
+
+    if (sub_020363A0() || sub_020593CC(CommSys_CurNetId()) != 0) {
+        return FALSE;
+    }
+
+    ov23_02242814();
+
+    u8 interactCheck = FALSE;
+
+    if (input->interact) {
+        ov23_02242830(interactCheck);
+        return FALSE;
+    }
+
+    if (input->menu) {
+        ov23_022427F8();
+        return FALSE;
+    }
+
+    if (ov23_02242458()) {
+        return FALSE;
+    }
+
+    return FALSE;
+}
+
+BOOL FieldInput_Process_Colosseum (FieldInput *input, FieldSystem *fieldSystem)
+{
+    if (input->mapTransition && 
+        input->transitionDir == DIR_SOUTH && 
+        sub_0205DB1C(Field_CurrentTileBehavior(fieldSystem))) {
+
+        sub_0203E880(fieldSystem, 9101, NULL);
+        return TRUE;
+    }
+
+    if (sub_020363A0() || sub_020593CC(CommSys_CurNetId()) != 0 || !sub_02059D2C()) {
+        return FALSE;
+    }
+
+    if (input->interact) {
+        MapObject *object;
+
+        if (sub_0203CA40(fieldSystem, &object) == TRUE && MapObject_GetMoveCode(object) != 0x1) {
+            if (sub_0205F588(fieldSystem->playerAvatar) == TRUE) {
+                sub_0205F5E4(fieldSystem->playerAvatar, PlayerAvatar_GetDir(fieldSystem->playerAvatar));
+            }
+
+            sub_0203E880(fieldSystem, sub_02062960(object), object);
+            return TRUE;
+        }
+    }
+
+    if (input->interact) {
+        sub_0205B2D4(fieldSystem);
+        return TRUE;
+    }
+
+    if (input->menu) {
+        Sound_PlayEffect(SEQ_SE_DP_WIN_OPEN);
+        sub_0203AABC(fieldSystem);
+        return TRUE;
+    }
+
+    return FALSE;
+}
+
+static int Field_CheckTrainerInfo (void)
+{
+    int netId;
+    TrainerInfo *info;
+
+    int ret = 0;
+
+    for (netId = 1; netId < 5; netId++) {
+        info = CommInfo_TrainerInfo(netId);
```

Comment:
```suggestion
    int ret = 0;

    for (int netId = 1; netId < 5; netId++) {
        TrainerInfo *info = CommInfo_TrainerInfo(netId);
```

Reply:
Bump

Reply:
Same as the other similar example in `Field_CalculateFriendship()`, can't get `netId` to match unless it's declared before `ret`, but `info` can be declared C99-style. Changing that, and I'll leave a comment about the other per Lhea's suggestion on `Field_CalculateFriendship()`.